### PR TITLE
Simplify convertible parameters

### LINKED ISSUE #677: Consider improving generic param names
Most functions generated are generic over the parameters to facilitate taking multiple types of arguments. In the docs this currently looks like this:

```rust
pub unsafe fn SomeFunction<'a, T1__: IntoParam<'a, SomeType>>(some_param: T1__);
```

While some of this complexity is unavoidable, it might help to give generic params more descriptive name. What strikes the right balance between not being overly verbose and providing proper guidance is tough. The best I can think of is `Param1`, but that might be too verbose.

### PR DESCRIPTION
**Pull request by kennykerr**
Fixes #677

### CODE REVIEW

**Review by rylev — CHANGES_REQUESTED** [review]
Can you update the docs/FAQ with the new syntax?

File: src/bindings.rs [hunk 1]
```diff
@@ -1747,11 +1747,8 @@ pub mod Windows {
             clippy::all
         )]
         pub mod Automation {
-            pub unsafe fn SysAllocStringLen<
-                'a,
-                T0__: ::windows::IntoParam<'a, super::SystemServices::PWSTR>,
-            >(
-                strin: T0__,
+            pub unsafe fn SysAllocStringLen<'a>(
+                strin: impl ::windows::IntoParam<'a, super::SystemServices::PWSTR>,
```
Comment (rylev) [inline src/bindings.rs:1751]: fyi: the param name should probably be `string` not `strin`

Comment (kennykerr) [reply src/bindings.rs:1751]: The name is derived from metadata. It is originally `strIn` and converted to lowercase by the windows crate.

**Comment by kennykerr** [comment]
Docs are updated.

**Pull request merged**